plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -53,6 +53,37 @@
  * The Azkaban adaptor for running a Spark Submit job.
  * Use this in conjunction with  {@link azkaban.jobtype.HadoopSecureSparkWrapper}
  *
+ * This class is used by azkaban executor to build the classpath, main args, env and jvm props
+ * for HadoopSecureSparkWrapper. Executor will then launch the job process and run 
+ * HadoopSecureSparkWrapper. HadoopSecureSparkWrapper will be the Spark client wrapper
+ * that uses the main args to launch spark-submit.
+ *
+ * Expect the following jobtype property:
+ * 
+ * spark.home (client default SPARK_HOME if user doesn't give a spark.version)
+ *             Conf will be either SPARK_CONF_DIR(we do not override it) or {spark.home}/conf
+ *
+ * spark.1.6.0.home (spark.{version}.home is REQUIRED for the {version} that we want to support.
+ *                  e.g. user can use spark 1.6.0 by setting spark.home=1.6.0 in their job property.
+ *                  This class will then look for plugin property spark.1.6.0.home to get the proper spark
+ *                  bin/conf to launch the client)
+ *
+ * spark.1.6.0.conf (OPTIONAL. spark.{version}.conf is the conf used for the {version}. 
+ *                  If not specified, the conf of this {version} will be spark.{version}.home/conf
+ *
+ * spark.dynamic.res.alloc.enabled (set to true if we want to enforce dynamic resource allocation policy.
```

Comment:
Instead of "spark.dynamic.res.alloc.enabled", do you think it's better to rename this config key to "spark.dynamic.res.alloc.enforced"?